### PR TITLE
feat: Persons Modal V2

### LINKED ISSUE #11480: styling on insight persons modal is off on chrome
## Bug description

scroll bars currently display on insight person modal, interfering with recordings. This used to render properly

image

*Please describe.*  
*If this affects the front-end, screenshots would be of great help.*  

## How to reproduce

1. on latest chrome build, open insight person modal. happ[ens regardless of zoom
2.
3.

## Environment

- [x] PostHog Cloud
- [ ] self-hosted PostHog, version/commit: please provide

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

Related to https://github.com/PostHog/posthog/issues/11352

Closes https://github.com/PostHog/posthog/issues/11480

The PersonsModal is an unpleasant experience and has a lot of complexity that can potentially be removed.

## Changes

* New PersonsModal behind a feature flag with a simplified external api taking only urls and a title.
* Muuuuuuch nicer UX in terms of jankiness of loading results, searching
* Improves a bunch of missing options around dropdown selects for example
* Fixes breakdown options for Lifecycle
* Far less buggy interface
* Some small non-improvements like using the "Actor" label from the results, which means empty results could say "No results" rather than "no persons". The simplified code felt worth the trade off.
* No more duplicated need to put the modal in the heirachy somewhere, it is not rendered in its own Portal.

|Before|After|
|----|----|
|Screenshot 2022-09-01 at 15 51 12|Screenshot 2022-09-01 at 15 51 09|
|Screenshot 2022-09-01 at 15 53 14|Screenshot 2022-09-01 at 15 52 03|

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

**Comment by neilkakkar** [comment]
Another related request: On a funnel / trend DAU chart, let the person modal number show the trend / funnel count, instead of the count of persons on the _current_ page?

**Comment by benjackwhite** [comment]
> Another related request: On a funnel / trend DAU chart, let the person modal number show the trend / funnel count, instead of the count of persons on the _current_ page?

That's a mouthful 😅  Could you do me a screenshot or something to give me a more practical view of what you mean? 😉

**Comment by neilkakkar** [comment]
Sorry 😅  , when you open any random modal, it shows `100 unique people`

image

when in fact, the trend can have 200k people. This is because the `limit` is usually 100, so only 100 people are loaded at a time.

**Comment by neilkakkar** [comment]
It's a hard-ish problem, since

1. We don't want to run the same query again to get the number of unique persons.
2. Showing the unique number of persons when we know it (any funnel, trend DAU chart) works, but might feel inconsistent since other charts (like trend total count) don't have the right value.
3. Some persons can be 'lost' with person-on-events (see https://github.com/PostHog/posthog/issues/11488)

I'm leaning a bit towards just doing (2), since it covers a decent chunk of use-cases, even though it's not consistent. 🤷

**Comment by benjackwhite** [comment]
> It's a hard-ish problem, since
> 
>     1. We don't want to run the same query again to get the number of unique persons.
> 
>     2. Showing the unique number of persons when we know it (any funnel, trend DAU chart) works, but might feel inconsistent since other charts (like trend total count) don't have the right value.
> 
>     3. Some persons can be 'lost' with person-on-events (see [Orphaned users for persons on events #11488](https://github.com/PostHog/posthog/issues/11488))
> 
> 
> I'm leaning a bit towards just doing (2), since it covers a decent chunk of use-cases, even though it's not consistent. 🤷

Yup this is exactly what my investigating realised. I think we do option 2 with an optional "knownCount" prop where we use that, otherwise we display something like "More than N unique persons"

File: tailwind.config.js [hunk 1]
```diff
@@ -163,7 +163,7 @@ module.exports = {
         // 'outlineOffset', // The outline-offset utilities like outline-offset-2
         // 'outlineStyle', // The outline-style utilities like outline-dashed
         // 'outlineWidth', // The outline-width utilities like outline-2
-        // 'overflow', // The overflow utilities like overflow-x-hidden
+        'overflow', // The overflow utilities like overflow-x-hidden
```
Comment (mariusandra) [inline tailwind.config.js:166]: What does this do?

**Review by mariusandra — APPROVED** [review]
Had a look through, found some things to nit, but it seems great overall!

It's weird to have two modals showing, but I understand the need to test. I don't see anything blocking code-wise for this to reach production, especially since its flagged. Will defer the interface review to @clarkus , though that might also be easier after this goes live.

**Comment by benjackwhite** [comment]
FYI @clarkus - didn't put you up for review as this is going in behind a flag as I want to test the technical part more heavily before actually showing it to anybody. There will be a follow up PR to properly release it where I definitely will tag you for review but if you have any comments for me to address in a follow up PR already, feel free to add them 👍

**Pull request merged**